src/index.js

```diff
@@ -3,22 +3,39 @@
  */
 
 // example of generating HTML with js
-function renderSidebar() {
-  const sidebarContents = `
-    <h2 class="email-header">Inbox</h2>
-    <ul class="email-list">
-      <li>
-        <button class="email-item" type="button">
-          <div class="sender-details">
-            <p>Email sender</p>
-            <span>Timestamp</span>
-          </div>
-          <p class="email-subject">Email1 subject</p>
-          <p>Email snippet</p>
-        </button>
-      </li>
-    </ul>
-  `;
+import store, {getMessageSubject, getMessageSender, formatSender, formatTimestamp} from './store'
+
+function renderThreads() {
+  const { threadIds } = store.mailboxes.INBOX;
+  const { messages } = store;
+
+  if (store.mailboxes.INBOX == null) return;
+
+  return threadIds.map( ind => {
+    const subject = getMessageSubject(messages[ind]);
+    const snippet = messages[ind].snippet;
+    const sender = getMessageSender(messages[ind]);
+    const timestamp = formatTimestamp(messages[ind].internalDate);
+
+    return `<li>
+    <button class="email-item" type="button">
+        <div class="sender-details">
+          <p class="email">${formatSender(sender)}</p>
+          <span class="time">${timestamp}</span>
+        </div>
+        <p class="email-subject">${subject}</p>
+        <p class="snippet">${snippet}</p>
+      </button>
+      </li>`;
+  }).join('');
+}
```

Comment:
I like the data flow here -- simple and unidirectional.  I also like how you're mapping from thread IDs to rendered `<li>`s.  You may need to break this out as your app gets more complex, but for now this an extremely clean way to do this.